postgres-driver/postgres_driver.go

```diff
@@ -0,0 +1,150 @@
+package postgresdriver
+
+import (
+	"database/sql"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/lib/pq"
+	indexer "github.com/pokt-foundation/pocket-indexer-lib"
+)
+
+const (
+	insertTransactionsScript = `
+	INSERT into transactions (hash, from_address, to_address, app_pub_key, blockchains, message_type, height, index, proof, stdtx, tx_result, tx, entropy, fee, fee_denomination)
+	VALUES (:hash, :from_address, :to_address, :app_pub_key, :blockchains, :message_type, :height, :index, :proof, :stdtx, :tx_result, :tx, :entropy, :fee, :fee_denomination)`
+	selectAllTransactionsScript   = "SELECT * FROM transactions"
+	selectTransactionByHashScript = "SELECT * FROM transactions WHERE hash = $1"
+)
+
+// PostgresDriver struct handler for PostgresDB related functions
+type PostgresDriver struct {
+	*sqlx.DB
+}
+
+// NewPostgresDriverFromConnectionString returns PostgresDriver instance from connection string
+func NewPostgresDriverFromConnectionString(connectionString string) (*PostgresDriver, error) {
+	db, err := sqlx.Open("postgres", connectionString)
+	if err != nil {
+		return nil, err
+	}
+
+	return &PostgresDriver{
+		DB: db,
+	}, nil
+}
+
+// NewPostgresDriverFromSQLDBInstance returns PostgresDriver instance from sdl.DB instance
+// mostly used for mocking tests
+func NewPostgresDriverFromSQLDBInstance(db *sql.DB) *PostgresDriver {
+	return &PostgresDriver{
+		DB: sqlx.NewDb(db, "postgres"),
+	}
+}
+
+type transaction struct {
```

Comment:
Don't we have the same struct in another file? Why resort to duplication?

Reply:
Right, I see that this is for the internal state in the DB. This should be clear in naming and in inline docs ideally.